pkg/fqdn/restore/restore.go

```diff
@@ -21,20 +20,24 @@ type IPRules []IPRule
 
 // IPRule stores the allowed destination IPs for a DNS names matching a regex
 type IPRule struct {
-	Re  RuleRegex
+	// v1.12 and older versions depend on having this value provided as a string, but for future
+	// improvements we gracefully handle reading it as nil as well. We always ensure that we currently
+	// never write nil in order to break rolling back to v1.12 or earlier.
+	Re  *string
```

Comment:
Have you tested downgrading from this version and validated that the rules are restored properly and traffic is not disrupted?

Reply:
Ended up "reverting" this change and keeping "99%" of the old logic here to ensure we behave the same way.

The actual disk format is exactly the same, so are the patterns/regexes.

I guess we can write some more tests, tho. I think it should be ok covered by `TestRestoredEndpoint` and `TestFullPathDependence`.